compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/ClosureDesugar.java

```diff
@@ -972,14 +972,12 @@ public void visit(BLangXMLElementLiteral xmlElementLiteral) {
 
     @Override
     public void visit(BLangXMLTextLiteral xmlTextLiteral) {
-        xmlTextLiteral.textFragments.forEach(this::rewriteExpr);
         xmlTextLiteral.concatExpr = rewriteExpr(xmlTextLiteral.concatExpr);
         result = xmlTextLiteral;
     }
 
     @Override
     public void visit(BLangXMLCommentLiteral xmlCommentLiteral) {
-        xmlCommentLiteral.textFragments.forEach(this::rewriteExpr);
```

Comment:
Shouldn't `concatExpr` be rewritten?

The following still crashes.

```ballerina
public function main() {
   int i = 1;
   xml _ = xml `<Books>${xml `<Book><!--${let function () returns int y = (() => i) in y()}--></Book>`}</Books>`;
}
```

Let's add tests like this for `BLangXMLCommentLiteral`, `BLangXMLTextLiteral`, and `BLangXMLQuotedString`.

Reply:
It should rewrite. Will fix it